syntax = "proto3";

package models;

message RolePermission {
  int32 role_id = 1;
  int32 permission_id = 2;
}

service RolePermissionService {
  rpc CreateRolePermission (CreateRolePermissionRequest) returns (RolePermission);
  rpc GetRolePermissionsByRoleId (GetRolePermissionsByRoleIdRequest) returns (RolePermissionsResponse);
}

message CreateRolePermissionRequest {
  int32 role_id = 1;
  int32 permission_id = 2;
}

message GetRolePermissionsByRoleIdRequest {
  int32 role_id = 1;
}

message RolePermissionsResponse {
  repeated RolePermission rolePermissions = 1;
}
